lib/internal/source_map/source_map_cache.js

```diff
@@ -351,11 +352,53 @@ function findSourceMap(sourceURL) {
   return sourceMap;
 }
 
+/**
+ *
+ * @param {Record<string, unknown>} callSite // The call site object to reconstruct from source map
+ * @returns {Record<string, unknown> | undefined} // The reconstructed call site object
+ */
+function reconstructCallSite(callSite) {
+  const { scriptName, lineNumber, column } = callSite;
+  const sourceMap = findSourceMap(scriptName);
+  if (!sourceMap) return;
+  const entry = sourceMap.findEntry(lineNumber - 1, column - 1);
+  if (!entry?.originalSource) return;
+  return {
+    __proto__: null,
+    functionName: callSite.name,
```

Comment:
Isn't this supposed to be `entry.name`?

Reply:
apparently `entry.name` is always undefined 🤔

Reply:
So with esbuild I was able to generate a sourcemap with names and it seems to be picking it up correctly